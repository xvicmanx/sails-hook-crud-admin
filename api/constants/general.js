const SALT_ROUNDS = 10;

module.exports = {
  SALT_ROUNDS,
};
