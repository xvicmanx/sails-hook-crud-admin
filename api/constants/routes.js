module.exports = {
  INDEX: 'GET /administrator',
  CLIENT_JS: 'GET /administrator/client.js',
  MODEL_COUNT: 'POST /administrator/model-count',
  MODEL_CREATE: 'POST /administrator/model-create',
  MODEL_DELETE: 'DELETE /administrator/model-delete/:id',
  MODEL_UPDATE: 'PUT /administrator/model-update/:id',
  MODEL_SEARCH: 'POST /administrator/model-search',
  MODEL_SEARCH_ALL: 'GET /administrator/model-search-all',
  MODELS_COUNT: 'GET /administrator/all-models-count',
  LOGIN: 'POST /administrator/login',
  UPLOAD: 'POST /administrator/model-upload-asset',
  MODELS_ASSETS: 'POST /administrator/models-assets',
  VIEW_CONTENT: 'POST /administrator/view-content',
  REPORT_CONTENT: 'GET /administrator/report-content',
  ASSET: 'GET /administrator/crud-asset/:hash',
};
