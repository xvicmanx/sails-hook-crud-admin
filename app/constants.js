import { getLabel, getButtonText } from './helpers/config';

const CRUD_MODELS = [
  'crudaction',
  'crudresource',
  'crudright',
  'crudgroup',
  'cruduser',
  'crudasset',
];

const Constants = {
  BUTTONS: {
    CREATE: getButtonText('create', 'Create'),
    UPDATE: getButtonText('update', 'Update'),
    REMOVE: getButtonText('remove', 'Remove'),
    SEE_DETAILS: getButtonText('seeDetails', 'See details')
  },
  LABELS: {
    HOME: getLabel('home', 'Home'),
    LOGOUT: getLabel('logout', 'Logout'),
    PERMISSIONS: getLabel('permissions', 'Permissions'),
    ASSETS: getLabel('assets', 'Assets'),
    ACTIONS: getLabel('actions', 'Actions'),
    RESOURCES: getLabel('resources', 'Resources'),
    WELCOME: getLabel('welcome', 'Welcome'),
    CREATE_FORM_TITLE: getLabel('createFormTitle', 'Create Item'),
    UPDATE_FORM_TITLE: getLabel('updateFormTitle', 'Update Item'),
    REMOVE_FORM_TITLE: getLabel('removeFormTitle', 'Remove existing Item'),
    CREATE_FORM_MESSAGE: getLabel('createFormMessage', 'Create a new item'),
    UPDATE_FORM_MESSAGE: getLabel('updateFormMessage', 'Update an existing item'),
    REMOVE_FORM_MESSAGE: getLabel('removeFormMessage', 'Are you sure you want to remove the item?'),
  },
  CRUD_MODELS,
  ASSETS_TYPES: {
    FILE: 'file',
    PICTURE: 'picture',
  },
};

export default Constants;
