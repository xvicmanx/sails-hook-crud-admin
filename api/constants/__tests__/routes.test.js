import constants from '../routes';

describe('api::constants::routes', () => {
  test('The url values are correct', () => {
    expect(constants).toMatchSnapshot();
  });
});
