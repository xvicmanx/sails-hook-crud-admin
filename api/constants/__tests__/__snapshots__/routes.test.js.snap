// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`api::constants::routes The url values are correct 1`] = `
Object {
  "ASSET": "GET /administrator/crud-asset/:hash",
  "CLIENT_JS": "GET /administrator/client.js",
  "INDEX": "GET /administrator",
  "LOGIN": "POST /administrator/login",
  "MODELS_ASSETS": "POST /administrator/models-assets",
  "MODELS_COUNT": "GET /administrator/all-models-count",
  "MODEL_COUNT": "POST /administrator/model-count",
  "MODEL_CREATE": "POST /administrator/model-create",
  "MODEL_DELETE": "DELETE /administrator/model-delete/:id",
  "MODEL_SEARCH": "POST /administrator/model-search",
  "MODEL_SEARCH_ALL": "GET /administrator/model-search-all",
  "MODEL_UPDATE": "PUT /administrator/model-update/:id",
  "UPLOAD": "POST /administrator/model-upload-asset",
  "VIEW_CONTENT": "POST /administrator/view-content",
}
`;
